clang-tidy reports uninitialized fields in format_string_checker

clang-tidy reports uninitialized fields in format_string_checker when building with `-std=c++20`.

Minimal example on https://godbolt.org/z/o79qcj5oo :

```cpp
#include <fmt/core.h>

std::string f()
{
    return fmt::format("hello world");
}
```

```
$ clang-tidy --extra-arg -Ifmt/include --extra-arg -std=c++20 test.cpp
...
fmt/include/fmt/core.h:2610:15: warning: 5 uninitialized fields at the end of the constructor call [clang-analyzer-optin.cplusplus.UninitializedObject]
        types_{mapped_type_constant<Args, buffer_context<Char>>::value...} {}
              ^
/tmp/example/fmt/include/fmt/core.h:725:7: note: uninitialized field 'this->context_.num_args_'
  int num_args_;
      ^~~~~~~~~
/tmp/example/fmt/include/fmt/core.h:726:15: note: uninitialized pointer 'this->context_.types_'
  const type* types_;
              ^~~~~~
/tmp/example/fmt/include/fmt/core.h:399:15: note: uninitialized pointer 'this->context_.basic_format_parse_context::format_str_.data_'
  const Char* data_;
              ^~~~~
/tmp/example/fmt/include/fmt/core.h:400:10: note: uninitialized field 'this->context_.basic_format_parse_context::format_str_.size_'
  size_t size_;
         ^~~~~
/tmp/example/fmt/include/fmt/core.h:655:7: note: uninitialized field 'this->context_.basic_format_parse_context::next_arg_id_'
  int next_arg_id_;
      ^~~~~~~~~~~~
/tmp/example/test.cpp:5:24: note: Calling constructor for 'basic_format_string<char, >'
    return fmt::format("hello world");
                       ^~~~~~~~~~~~~
/tmp/example/fmt/include/fmt/core.h:2733:5: note: Taking true branch
    if constexpr (detail::count_named_args<Args...>() ==
    ^
/tmp/example/fmt/include/fmt/core.h:2737:47: note: Calling constructor for 'format_string_checker<char, >'
      detail::parse_format_string<true>(str_, checker(s));
                                              ^~~~~~~~~~
/tmp/example/fmt/include/fmt/core.h:2610:15: note: 5 uninitialized fields at the end of the constructor call
        types_{mapped_type_constant<Args, buffer_context<Char>>::value...} {}
              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

fmtlib version: latest master at time of writing (661b23edeb52d400cf5812e7330f14f05c072fab), but as far as I can see this affects all versions after June 2022 (b135f1c01449f6f41be3933437797e768b54295a).

I am getting this error with clang-tidy 15.0.7 on Linux but the same error also occurs on the clang-tidy in compiler explorer (linked above).

The fix seems to be relatively straightforward, I will submit a pull request.

--- a/include/fmt/core.h
+++ b/include/fmt/core.h
@@ -2599,15 +2599,15 @@ template <typename Char, typename... Args> class format_string_checker {
   // needed for compile-time checks: https://godbolt.org/z/GvWzcTjh1.
   using parse_func = const Char* (*)(parse_context_type&);
 
+  type types_[num_args > 0 ? static_cast<size_t>(num_args) : 1];
   parse_context_type context_;
   parse_func parse_funcs_[num_args > 0 ? static_cast<size_t>(num_args) : 1];
-  type types_[num_args > 0 ? static_cast<size_t>(num_args) : 1];
 
  public:
   explicit FMT_CONSTEXPR format_string_checker(basic_string_view<Char> fmt)
-      : context_(fmt, num_args, types_),
-        parse_funcs_{&parse_format_specs<Args, parse_context_type>...},
-        types_{mapped_type_constant<Args, buffer_context<Char>>::value...} {}
+      : types_{mapped_type_constant<Args, buffer_context<Char>>::value...},
+        context_(fmt, num_args, types_),
+        parse_funcs_{&parse_format_specs<Args, parse_context_type>...} {}
 
   FMT_CONSTEXPR void on_text(const Char*, const Char*) {}
 

